lib/pure/math.nim

```diff
@@ -141,6 +141,22 @@ when not defined(JS): # C
   proc ln*(x: float32): float32 {.importc: "logf", header: "<math.h>".}
   proc ln*(x: float64): float64 {.importc: "log", header: "<math.h>".}
     ## Computes the natural log of `x`
+else: # JS
+  proc sqrt*(x: float32): float32 {.importc: "Math.sqrt", nodecl.}
+  proc sqrt*(x: float64): float64 {.importc: "Math.sqrt", nodecl.}
+
+  proc ln*(x: float32): float32 {.importc: "Math.log", nodecl.}
+  proc ln*(x: float64): float64 {.importc: "Math.log", nodecl.}
+
+proc log*[B, X: SomeFloat](b: B, x: X): auto =
```

Comment:
Is there a precedent to keep base as the first argument?  To me, it feels natural to have base as the second argument, with a default so the not using the second argument defaults `log` to do `ln`. 

So the usage would be:

```
x = log(y, 2)
x = log(y, 10)
x = log(y)
```

Precedents for above proposal:

- [Python][1]
- [Emacs lisp][2]

[1]: https://docs.python.org/3/library/math.html#power-and-logarithmic-functions
[2]: https://www.gnu.org/software/emacs/manual/html_node/elisp/Math-Functions.html

Reply:
I used from Julia's precedent.
- [Julia](https://docs.julialang.org/en/stable/stdlib/math/#Base.log-Tuple{Any})

Reply:
Found few more precedents for *log(arg, base)*:

- [Common Lisp][1]
- [Rust][2]

That said, this is not a big deal.. I'm just giving my perspective. I actually found the order reversed when I read this commit (I am learning Nim by reading commits :-)). And then I realized that in all the languages I have ever used, the base arg was optional and came second, or it was completely absent (in which case I would do *log(foo)/log(base)* manually.

So it just boils down to which set of precendents gets higher weight and is subjective to of course you and Nim devs.

[1]: http://clhs.lisp.se/Body/f_log.htm
[2]: https://doc.rust-lang.org/std/primitive.f64.html#method.log

Reply:
On a different note, I see that you added `ln` too.. so would `log(math.E, *somefloat*)` and `ln(*somefloat*)` be always the same? Because I see that both functions should return the exact same result, but they are implemented differently (one is using the C `log` vs other is a float division in Nim).

Or, should the `log` contain a case statement that calls `ln` if the `b` arg is `math.E`? probably not? I am just throwing out ideas.

Reply:
I'm in favor of `log(arg, base)`.

Reply:
IMO, I have no specific favor for log(base, x) or log(x, base). The first one is close to the mathematical representation, and the second one is natural for programming because often the second argument is optional.
I didn't add `ln`. I have only moved the place of definition. I like `log` for natural log than `ln`, but the current Nim adopts `ln`.

Reply:
Many languages adopts ``log(x, base)`` and at least 2 people likes it. I change it to it.